jax/core.py

```diff
@@ -1705,14 +1705,48 @@ def shadowed_subst(name):
     new_params[name] = subst_axis_names_jaxpr(jaxpr, shadowed_subst)
   return new_params
 
+class DuplicateAxisNameError(Exception):
+  def __init__(self, var):
+    self.var = var
+    self.eqn = None
+
+def subst_axis_names_var(v: Var, subst: AxisSubst, var_map: Dict[Var, Var]):
+  # Var identity is load-bearing, so we can't have duplicates!
+  if v is unitvar: return v
+  if v is dropvar: return v
+  assert v not in var_map
+  if not hasattr(v.aval, 'named_shape'):
+    var_map[v] = v
+    return v
+  names = sum([subst(name) for name in v.aval.named_shape], ())
```

Comment:
```suggestion
  names = tuple(it.chain.from_iterable(subst(name) for name in v.aval.named_shape))
```

Reply:
What kind of code golf is that? My version was shorter 😛

Reply:
Ah true! But it's efficiency golf: I think the `sum` creates more temporary tuples (one for each pair of tuples added).